### PR TITLE
Add unsafe-regex rule

### PR DESCRIPTION
**Pull request by bdougherty**
For sindresorhus/xo#227

### CODE REVIEW

File: docs/rules/unsafe-regex.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Detect unsafe regular expressions
```
Comment (sindresorhus) [inline docs/rules/unsafe-regex.md]: Let's use the word `disallow` instead of `detect`. In the other places too.

File: index.js [hunk 2]
```diff
@@ -33,7 +33,8 @@ module.exports = {
 				'unicorn/new-for-builtins': 'error',
 				'unicorn/regex-shorthand': 'error',
 				'unicorn/prefer-spread': 'error',
-				'unicorn/error-message': 'error'
+				'unicorn/error-message': 'error',
+				'unicorn/unsafe-regex': 'error'
```
Comment (sindresorhus) [inline index.js]: `unsafe-regex` sounds like we enforce unsafe regexes. I think it should be `no-unsafe-regex`.

File: rules/unsafe-regex.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	};
+};
+
+module.exports = {
+	create,
+	meta: {
+		docs: {
+			url: getDocsUrl()
+		},
+		fixable: 'code'
```
Comment (sindresorhus) [inline rules/unsafe-regex.js]: It's not fixable, so this should be removed.

**Comment by bdougherty** [comment]
Should be good now!

**Comment by sindresorhus** [comment]
Looks great! Thank you :)

**Pull request merged**